### PR TITLE
[civ2][flakiness/2] Migrate core medium python tests to civ02

### PR DESCRIPTION
**Pull request by can-anyscale**
Migrate core medium python tests to civ2. Concretely
- Splitting the existing medium step to run non-core tests + flaky-core tests
- Move non-flaky-core tests to civ2
- Also remove the 'kubernetes' tag filters since this tag no longer exist

Tests:
- CI

### CODE REVIEW

**Review by aslonnie — COMMENTED** [review]
seems to depend on the 1st part?

File: .buildkite/pipeline.build_core.yml
Comment (aslonnie) [inline .buildkite/pipeline.build_core.yml:4]: why is this line removed?

File: .buildkite/pipeline.build_core.yml
Comment (aslonnie) [inline .buildkite/pipeline.build_core.yml:22]: try with no sed?

File: .buildkite/pipeline.build_core.yml
Comment (can-anyscale) [reply .buildkite/pipeline.build_core.yml]: i was removing it for ci to trigger the job for testing, added back

**Review by aslonnie — APPROVED** [review]
lgtm, not sure if you want some one from core team to have a look too.

**Pull request merged**
